src/PowerShellEditorServices.Host/CodeLens/PesterCodeLensProvider.cs

```diff
@@ -15,71 +15,90 @@ namespace Microsoft.PowerShell.EditorServices.CodeLenses
-                        pesterSymbol.TestName,
-                    }),
+                new CodeLens(
+                    this,
+                    scriptFile,
+                    pesterSymbol.ScriptRegion,
+                    new ClientCommand(
+                        "PowerShell.RunPesterTests",
+                        "Run tests",
+                        new object[] { scriptFile.ClientFilePath, false /* No debug */, pesterSymbol.TestName })),
 
-                new ClientCommand(
-                    "PowerShell.RunPesterTests",
-                    "Debug tests",
-                    new object[]
-                    {
-                        scriptFile.ClientFilePath,
-                        true, // Run in debugger
-                        pesterSymbol.TestName,
-                    }),
+                new CodeLens(
+                    this,
+                    scriptFile,
+                    pesterSymbol.ScriptRegion,
+                    new ClientCommand(
+                        "PowerShell.RunPesterTests",
+                        "Debug tests",
+                        new object[] { scriptFile.ClientFilePath, true /* Run in debugger */, pesterSymbol.TestName })),
             };
 
-            return
-                clientCommands.Select(
-                    command =>
-                        new CodeLens(
-                            this,
-                            scriptFile,
-                            pesterSymbol.ScriptRegion,
-                            command));
+            return codeLensResults;
         }
 
+        /// <summary>
+        /// Get all Pester CodeLenses for a given script file.
+        /// </summary>
+        /// <param name="scriptFile">The script file to get Pester CodeLenses for.</param>
+        /// <returns>All Pester CodeLenses for the given script file.</returns>
         public CodeLens[] ProvideCodeLenses(ScriptFile scriptFile)
         {
-            var symbols =
-                this.symbolProvider
-                    .ProvideDocumentSymbols(scriptFile);
+            var lenses = new List<CodeLens>();
+            foreach (SymbolReference symbol in _symbolProvider.ProvideDocumentSymbols(scriptFile))
+            {
+                if (symbol is PesterSymbolReference pesterSymbol)
+                {
+                    if (pesterSymbol.Command != PesterCommandType.Describe)
+                    {
+                        continue;
+                    }
 
-            var lenses =
-                symbols
-                    .OfType<PesterSymbolReference>()
-                    .Where(s => s.Command == PesterCommandType.Describe)
-                    .SelectMany(s => this.GetPesterLens(s, scriptFile))
-                    .Where(codeLens => codeLens != null)
-                    .ToArray();
+                    lenses.AddRange(GetPesterLens(pesterSymbol, scriptFile));
+                }
+            }
 
-            return lenses;
+            return lenses.ToArray();
```

Comment:
echo: `SelectMany` vs `List > Array`

Reply:
Going to leave this one on the basis of avoiding some LINQ closures